### ISSUE TITLE
stimela fails on Fedora 31

### ISSUE THREAD

**Issue by adamdeller** [ratt-ru/Stimela#585]
Hi, I recently tried to rerun Stimela after upgrading from Fedora twenty-something (under which it ran peachily, with Docker and python2) to 31.  This promptly failed, because of Fedora 31's embrace of cgroup v2, which is incompatible with docker.  So I upgraded to the latest version of stimela and tried again with podman, but no luck.  Even stripping right back to a very basic test (exact copy/paste of the basic recipe example shown on https://github.com/ratt-ru/Stimela/wiki/Putting-together-a-basic-reduction-recipe) barfs as follows:

adeller@ar313-adeller trunk helloworld> stimela run test.py 
2020-04-28 12:50:44 STIMELA.Simulation Example INFO: ---------------------------------
2020-04-28 12:50:44 STIMELA.Simulation Example INFO: Stimela version 1.4.1
2020-04-28 12:50:44 STIMELA.Simulation Example INFO: Sphesihle Makhathini <sphemakh@gmail.com>
2020-04-28 12:50:44 STIMELA.Simulation Example INFO: Running: Simulation Example
2020-04-28 12:50:44 STIMELA.Simulation Example INFO: ---------------------------------
Traceback (most recent call last):
  File "/home/adeller/.local/bin/stimela", line 14, in <module>
    main.main([a for a in sys.argv[1:]])
  File "/home/adeller/.local/lib/python3.7/site-packages/stimela/main.py", line 643, in main
    _cmd(argv)
  File "/home/adeller/.local/lib/python3.7/site-packages/stimela/main.py", line 265, in run
    exec(stdr.read(), _globals)
  File "<string>", line 46, in <module>
  File "/home/adeller/.local/lib/python3.7/site-packages/stimela/recipe.py", line 475, in add
    time_out=time_out)
  File "/home/adeller/.local/lib/python3.7/site-packages/stimela/recipe.py", line 227, in setup_job
    cont.RUNSCRIPT, perm="ro")
  File "/home/adeller/.local/lib/python3.7/site-packages/stimela/podman.py", line 103, in add_volume
    "Directory {0} cannot be mounted on container: File doesn't exist".format(host))
OSError: Directory /usr/bin/stimela_runscript cannot be mounted on container: File doesn't exist

I tried to reproduce with udocker (it complained about add_volume not having the parameter perm, so I added that just to test) but same result.  And this is the same issue I ran into with my actual, non-trivial code I was trying to run.  Perhaps I'm just being thick, but is this a basic problem? Or is it potentially an issue with python3, as my previous install of stimela would have been using python2?  I did do the purge of ~/.stimela/ as suggested on https://github.com/ratt-ru/Stimela/wiki/Purging-stimela-from-my-system-and-rolling-between-versions.

Apologies in advance if this was better raised via another forum (is there a stimela users list?) rather than a github issue.

**Comment by o-smirnov** [comment]
@adamdeller, long time!

For completeness sake, have you tried with singularity? I believe docker/singularity is the core functionality for now, and podman is a beta-ish..

**Comment by adamdeller** [comment]
@o-smirnov indeed!

I hadn't tried singularity, but for completeness' sake (which sounds like an exasperated grandma exclamation - "Oh, for completeness' sake, Adam!") I did, but same end result:

  File "/home/adeller/.local/lib/python3.7/site-packages/stimela/singularity.py", line 82, in add_volume
    "Path {0} cannot be mounted on container: File doesn't exist".format(host))
OSError: Path /usr/bin/stimela_runscript cannot be mounted on container: File doesn't exist

**Comment by SpheMakh** [comment]
It looks like it's trying to access the stimela binary from the location. 

`OSError: Directory /usr/bin/stimela_runscript cannot be mounted on container: File doesn't exist` 

This is probably the offending lines. 

https://github.com/ratt-ru/Stimela/blob/e631033e1a67a46985fe5905c297756c84dd202b/stimela/recipe.py#L23 

https://github.com/ratt-ru/Stimela/blob/e631033e1a67a46985fe5905c297756c84dd202b/stimela/recipe.py#L226-L227

I should use `shutil.which` like I do here

https://github.com/ratt-ru/Stimela/blob/e631033e1a67a46985fe5905c297756c84dd202b/stimela/singularity.py#L13-L16

**Comment by adamdeller** [comment]
And yea, there was much rejoicing, as the correct path to stimela_runscript (obtained using which()) indeed solved the issue.  

Is it also fair to say that that error message is slightly misleading as currently written?

Thanks muchly for the help.

**Comment by adamdeller** [comment]
OK, one more thing, once I moved back from the basic recipe example to what I was actually trying to do, simms is barfing because it seems like the lon_lat parameter is not getting passed to simms properly.  I have the following configuration:

simms_params = {
   "msname"    :   MSNAME,
   "telescope" :   "askap",
   "type"      :   "ascii",
   "lon-lat-elv":  [116.6314243,-26.697, 361],
   "coord-sys" :   "itrf", # or enu, or wgs84
   "antenna-file": "askap-itrf.txt",
   "direction" : "J2000,12h00m00s,-40d00m00s",
   "synthesis" :   1.0/3600.0, # Synthesis time in hours
   "dtime"     :   1, # integration time in seconds
   "freq0"     :   "1100MHz", # Start frequency
   "dfreq"     :   "4MHz",
   "nchan"     :   84,
}

but each element of the list is getting passed to simms as a separate --lon-lat-elv parameter:

subprocess.CalledProcessError: Command '['simms', '--name', '/stimela_mount/msdir/askap-stimsim.ms', '--tel', 'askap', '--type', 'ascii', '--coord-sys', 'itrf', '--lon-lat-elv', '116.6314243', '--lon-lat-elv', '-26.697', '--lon-lat-elv', '361', '--direction', 'J2000,12h00m00s,-40d00m00s', '--synthesis', '0.000277777777778', '--dtime', '1', '--freq0', '1100MHz', '--dfreq', '4MHz', '--nband', '1', '--nchan', '84', '--pol', 'XX XY YX YY', '--feed', 'perfect X Y', '--nolog', '/stimela_mount/input/askap-itrf.txt']' returned non-zero exit status 1

Which of course leads ultimately to makems barfing:
#     makems(msname="/stimela_mount/msdir/askap-stimsim.ms", label="None", tel="askap", pos="/stimela_mount/input/askap-itrf.txt", pos_type="ascii", synthesis=0.0002778, scan_length=None, dtime="1.0", freq0=['1100MHz'], dfreq=['4MHz'], nchan=[84], stokes="XX XY YX YY", setlimits=False, elevation_limit=0.000000, shadow_limit=0.000000, coords="itrf",lon_lat="361", noup=False, nbands=1, direction=['J2000,12h00m00s,-40d00m00s'], outdir=".",date="None",fromknown=False, feed="perfect X Y",scan_lag=0,auto_corr=False,optimise_start=False)
#   File "/usr/local/lib/python2.7/dist-packages/simms/casasm.py", line 98, in makems
#     lon,lat = ['%sdeg'%i for i in tmp[:2]]
# ValueError: need more than 1 value to unpack

I can't quite track down where/how the lon-lat-elv parameter is mistakenly getting split and each element passed separately as a repeat argument to simms.  But this worked prior to my recent upgrade, so it must have been something introduced in the last year or two.

**Comment by adamdeller** [comment]
OK, I was able to bypass the problem with simms by modifying the stimela/cargo/cab/simms/parameters.json file, changing to "dtype": "str" for this parameter.  However, I compared this to my previously installed python2.7 version of stimela and saw that the dtype was the same there as the new version ( "dtype"     :   "list:float") so I'm not quite sure how that ever worked.

I then also found that the parameters.json file for the simulator cab is busted in quite a number of places, with a few "floats" changed to "ints" (most notably the gain parameters), and quite a few description strings seemingly mistakenly copy pasted (the gain min max parameters seemed to have the description strings for phases, or periods).

With a few hacks I think I've got it going again. Hope this feedback is useful, I could start it as a separate bug if you prefer so it would be easier to track.
